### PR TITLE
Xrefs select top item

### LINKED ISSUE #1363: X-Refs Widget Improvements: empty `Code Preview`
`Code preview` is empty after dialog appears
---

image

Probably we have to select first xref of any available type (xrefs **to** has to be checked first)

### PR DESCRIPTION
**Pull request by ncyellow**
Try to select first refs. If refs is empty try to select xrefs.

xrefs select

fixes #1363

### CODE REVIEW

File: src/common/Helpers.cpp [hunk 1]
```diff
@@ -72,6 +72,16 @@ QTreeWidgetItem *appendRow(QTreeWidget *tw, const QString &str, const QString &s
     return tempItem;
 }
 
+bool selectFirstItem(QTreeWidget* tw)
+{
+    if (tw->topLevelItem(0))
+    {
```
Comment (xarkes) [inline src/common/Helpers.cpp]: Better put that brace on the same line of the `if`

File: src/dialogs/XrefsDialog.cpp [hunk 2]
```diff
@@ -64,6 +64,13 @@ void XrefsDialog::fillRefs(QList<XrefDescription> refs, QList<XrefDescription> x
 
     // Adjust columns to content
     qhelpers::adjustColumns(ui->toTreeWidget, 0);
+
+    // try to select first item from refs or xrefs
+    if (!qhelpers::selectFirstItem(ui->toTreeWidget))
+    {
```
Comment (xarkes) [inline src/dialogs/XrefsDialog.cpp]: Same comment here.

File: src/common/Helpers.cpp (hunk 1 above)
Comment (ncyellow) [reply src/common/Helpers.cpp]: done

File: src/common/Helpers.cpp [hunk 3]
```diff
@@ -72,6 +72,15 @@ QTreeWidgetItem *appendRow(QTreeWidget *tw, const QString &str, const QString &s
     return tempItem;
 }
```
Comment (ITAYC0HEN) [inline src/common/Helpers.cpp]: Please document this function using our documentation formatting

**Comment by ITAYC0HEN** [comment]
Looks very good! Thank you :)
Document the function using our documentation style
https://cutter.re/docs/code.html#docstrings

File: src/common/Helpers.cpp (hunk 3 above)
Comment (ncyellow) [reply src/common/Helpers.cpp]: done

**Comment by xarkes** [comment]
Looks good to me, thanks!

**Pull request merged**